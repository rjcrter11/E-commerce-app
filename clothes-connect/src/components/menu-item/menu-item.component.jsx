import React from 'react';
import { withRouter } from 'react-router-dom';

import './menu-item.styles.scss'

export const MenuItem = ({ title, imageUrl, size, history, linkUrl, match }) => (

    <div
        data-testid="menu-item-container"
        className={`${size} menu-item`}
        onClick={() => history.push(`${match.url}${linkUrl}`)}  >
        <div data-testid="background-image-container" className='background-image' style={{
            backgroundImage: `url(${imageUrl})`
        }} />
        <div className='content' >
            <h1 className='title' >{title.toUpperCase()}</h1>
            <span className='subtitle' >SHOP NOW</span>
        </div>
    </div>
)


export default withRouter(MenuItem);