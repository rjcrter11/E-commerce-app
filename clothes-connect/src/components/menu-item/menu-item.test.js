import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import { MenuItem } from './menu-item.component'

describe("MenuItem component", () => {
    let wrapper;
    let enzymeWrapper;
    let mockMatch;
    let mockHistory;
    const linkUrl = '/hats';
    const size = 'large';
    const imageUrl = 'testImage';

    beforeEach(() => {
        mockMatch = {
            url: '/shop'
        };

        mockHistory = {
            push: jest.fn()
        };

        const mockProps = {
            match: mockMatch,
            history: mockHistory,
            linkUrl,
            size,
            title: 'hats',
            imageUrl
        };
        wrapper = create(<MenuItem {...mockProps} />)
        enzymeWrapper = shallow(<MenuItem {...mockProps} />)
    });

    it('should render MenuItem component', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    })
    it('should call history.push with the right string when MenuItemContainer is clicked', () => {
        enzymeWrapper.find('[data-testid="menu-item-container"]').simulate('click');

        expect(mockHistory.push).toHaveBeenCalled()
        expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
    })
    it.skip('should pass size to MenuItemContainer as the prop size', () => {
        expect(enzymeWrapper.find('[data-testid="menu-item-container"]').prop('size')).toBe(size)
    })
    it.skip('should pass imageUrl to backgroundImageContainer as the prop imageUrl', () => {
        expect(enzymeWrapper.find('[data-testid="background-image-container"]').prop('imageUrl')).toBe(imageUrl)
    })
})