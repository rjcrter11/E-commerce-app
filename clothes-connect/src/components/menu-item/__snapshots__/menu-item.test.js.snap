// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render MenuItem component 1`] = `
<div
  className="large menu-item"
  data-testid="menu-item-container"
  onClick={[Function]}
>
  <div
    className="background-image"
    data-testid="background-image-container"
    style={
      Object {
        "backgroundImage": "url(testImage)",
      }
    }
  />
  <div
    className="content"
  >
    <h1
      className="title"
    >
      HATS
    </h1>
    <span
      className="subtitle"
    >
      SHOP NOW
    </span>
  </div>
</div>
`;
